### PR TITLE
Replace anonymous inner classes with lambdas

### PR DESCRIPTION
**Pull request by Beacze**

### CODE REVIEW

File: Kitodo/src/main/java/de/sub/goobi/helper/Helper.java [hunk 1]
```diff
@@ -507,65 +500,58 @@ public static AktuelleSchritteForm getCurrentTaskForm() {
...
-            regexList.add(prefix + "\\.[jJ][pP][2]");
-            regexList.add(prefix + "\\.[pP][nN][gG]");
-            regexList.add(prefix + "\\.[gG][iI][fF]");
-
-            for (String regex : regexList) {
-                if (name.matches(regex)) {
-                    fileOk = true;
-                }
-            }
+    public static final FilenameFilter imageNameFilter = (dir, name) -> {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/de/sub/goobi/helper/Helper.java]: Missing JavaDoc for public method. Or move this code into its own class to get ride out of this inner class usage.

File: Kitodo/src/main/java/de/sub/goobi/helper/Helper.java [hunk 2]
```diff
@@ -507,65 +500,58 @@ public static AktuelleSchritteForm getCurrentTaskForm() {
...
-            regexList.add(prefix + "\\.xls");
-            regexList.add(prefix + "\\.xlsx");
-            regexList.add(prefix + "\\.pptx");
-            regexList.add(prefix + "\\.ppt");
-
-            for (String regex : regexList) {
-                if (name.matches(regex)) {
-                    fileOk = true;
-                }
+    public static final FilenameFilter dataFilter = (dir, name) -> {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/de/sub/goobi/helper/Helper.java]: Missing JavaDoc for public method. Or move this code into its own class to get ride out of this inner class usage.

File: Kitodo/src/main/java/de/sub/goobi/helper/tasks/EmptyTask.java [hunk 3]
```diff
@@ -58,20 +58,10 @@
...
-         * @see java.lang.Thread.UncaughtExceptionHandler#uncaughtException(java.lang.Thread,
-         *      java.lang.Throwable)
-         */
-        @Override
-        public void uncaughtException(Thread origin, Throwable exception) {
-            if (origin instanceof EmptyTask) {
-                EmptyTask task = (EmptyTask) origin;
-                task.setException(exception);
-            }
+    public static final Thread.UncaughtExceptionHandler CATCH_ALL = (origin, exception) -> {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/de/sub/goobi/helper/tasks/EmptyTask.java]: JavaDoc did not matched or move this code into its own class to get ride out of this inner class usage.

Comment (Beacze) [reply Kitodo/src/main/java/de/sub/goobi/helper/tasks/EmptyTask.java]: JavaDocs can't be made for lambdas - param attributes are not allowed there - and it is constant.

File: Kitodo/src/main/java/de/sub/goobi/helper/Helper.java (hunk 2 above)
Comment (Beacze) [reply Kitodo/src/main/java/de/sub/goobi/helper/Helper.java]: JavaDocs can't be made for lambdas - param attributes are not allowed there - and it is constant.

File: Kitodo/src/main/java/de/sub/goobi/helper/Helper.java (hunk 1 above)
Comment (Beacze) [reply Kitodo/src/main/java/de/sub/goobi/helper/Helper.java]: JavaDocs can't be made for lambdas - param attributes are not allowed there - and it is constant.

Comment (henning-gerhardt) [reply Kitodo/src/main/java/de/sub/goobi/helper/Helper.java]: I would move this code to its own classes with proper JavaDoc and explanation and add tests for this class to prove its correct implementaion. Maybe I'm a old school programmer but this is more readable then this lambda expression.

Comment (Beacze) [reply Kitodo/src/main/java/de/sub/goobi/helper/Helper.java]: I have simplified this two lambdas.

**Pull request merged**